UsingTheServer.md

```diff
@@ -0,0 +1,157 @@
+   
+buildscript {
+    repositories {
+        jcenter()
+    }
+    dependencies {
+        classpath 'com.github.jengelman.gradle.plugins:shadow:2.0.1'
+    }
+}
+   
+dependencies {
+    compile project(':core')
+    testCompile group: 'junit', name: 'junit', version: '4.12'
+}
+   
+repositories {
+    jcenter()
+}
+   
+jar {
+    manifest {
+        attributes(
+                'Main-Class': 'com.saralein.YOUR_APPLICATION.Main'
+        )
+    }
+}
+```
+
+## Setting up your application Main
+
+Usage of the server requires some setup within your application.
+
+1. Within `main` in your application `Main` class, you should specify a base location from which you plan to server files. In the cobspec application, this location is `user.dir`.  Note that the specific directory from which you plan to serve is appended via the `args` passed into `main`.
+2. The server `main` method requires the following arguments:
+    
+    `main(String[] args, String home, Function<Path, Routes> setupRoutes)`
+    
+    Pass your application `args` as the first argument and your base serving location as the second `home` argument.
+    
+    To satisfy the third argument, your application must pass a function to the server `main` which accepts a `Path` as its argument as returns an instance of the `server` `Routes` class.
+    
+### Routes
+
+Minimal setup of a `Routes` instance requires instantiation with four arguments:
+
+1. Routes hashmap - follows the structure `HashMap<String, HashMap<String, Controller>>`, where the key is your route URI and the associate hashmap is a method ("GET", "POST", "PUT", etc.) and the controller which handles the specific method for that URI.
+   
+   For example, if you wanted to have "GET" and "POST" requests for a "/board" route, your routes hashmap would be:
+   
+   ```
+   {"/board":
+     {
+       "GET": Controller,
+       "POST": Controller
+     }
+   }
+   ```
+2. Directory controller - handles directory requests, implements the server `Controller` interface
+3. File controller - handles file requests, implements the server `Controller` interface
+4. Not found controller - handles 404s on absent resources, implements the server `Controller` interface
+
+For ease of use, you may choose to follow the `ResponseBuilder` implementation in the cobspec application for creating your routes.
+
+### Controllers
+
+Your controllers should implement the server `Controller` interface:
+
+```java
+public interface Controller {
+    Response createResponse(Request request);
+}
+
+```
+
+For example controllers/responses, see the cobspec controllers.
+
+### Example Implementation
+
+With your function setup, simply pass `args`, `home`, and your function to the server `main` function.  Below is an example of how your main might look:
+
+```java
+public class Main {
+    public static void main(String[] args) {
+        String home = System.getProperty("user.dir");
+        com.saralein.core.Main.main(args, home, Main::setupRoutes);
```

Comment:
Seems like you may want to not use `Main`. Perhaps `Server` or something?

Reply:
Updated it `Main` in `server` to `HttpServer`.